### PR TITLE
use separate protocol versions for the different services

### PR DESCRIPTION
**Pull request by arvidn**
### Purpose:

When updating the harvester protocol, we should bump its version number. Currently all our protocols use a single protocol version constant, which means we would also have to bump the version number of the full node- and wallet protocol as well (even though they haven't changed).

This patch separates the constant into one version number per protocol.

### CODE REVIEW

**Review by aqk — DISMISSED** [review]

**Comment by harold-b** [comment]
I might be mistaken, but I don't see a place where the protocol version is validated on either end of the connection.`ws_connection.py.perform_handshake()` seems like it would be the ideal location for this check

**Comment by harold-b** [comment]
I pushed a commit to perform the check. Let me know if that works.
There was already an error message for it `INCOMPATIBLE_PROTOCOL_VERSION`, which did not seem to be used in the codebase.

File: chia/server/ws_connection.py [hunk 1]
```diff
@@ -223,6 +223,9 @@ async def perform_handshake(
             if inbound_handshake.network_id != network_id:
                 raise ProtocolError(Err.INCOMPATIBLE_NETWORK_ID)
 
+            if inbound_handshake.protocol_version != protocol_version:
+                raise ProtocolError(Err.INCOMPATIBLE_PROTOCOL_VERSION)
```
Comment (arvidn) [inline chia/server/ws_connection.py:228]: both the full node- and wallet protocols need to be backwards compatible, so I don't think it's safe to be this strict for all protocols.

For the full node- and wallet protocols, we use the version to know whether new messages are supported or not.

Comment (harold-b) [reply chia/server/ws_connection.py:228]: OK understood. Is it fine if we enforce this then only for the Harvester and Farmer protocols?

**Comment by harold-b** [comment]
Also, are handshakes legal bi-directionally? That is, can a `Harvester` initiate a connection to a `Farmer` and vice-versa or is it only ever uni-directional?

**Pull request closed without merging**